### ISSUE TITLE
Provide JS module

### ISSUE THREAD

**Issue by steren** [GoogleCloudPlatform/stackdriver-errors-js#2]
Installing via a script is not the preferred way to install dependencies in a modern web application.

Developers should be able to use the library by importing a module in their application.

We should be compatible with the common module conventions (AMD, CommonJS, JS modules?)

**Comment by oshalygin** [comment]
I'd love to get started on something like this, and with the help wanted label, are you still open to a PR?

**Comment by steren** [comment]
Yes of course, any help is appreciated. Feel free to submit a PR

**Comment by evanjmg** [comment]
Is there any usage instructions for npm? I can't use it as a module unless I know how to use it as such. A simple import did not work. Please add to readme

**Comment by mikelambert** [comment]
```
npm install github:GoogleCloudPlatform/stackdriver-errors-js
````

And then:
```
import { StackdriverErrorReporter } from 'stackdriver-errors-js';
```

works for me.

**Comment by steren** [comment]
Thanks. I suppose this is ES6 syntax right?
Feel free to send a PR to add this info to https://github.com/GoogleCloudPlatform/stackdriver-errors-js/blob/master/README.md#setup-for-javascript

**Comment by mikelambert** [comment]
Ideally you would 'npm publish' this repo up to the npm servers, and then repush it as needed with semver. Someone else could do that, but then that someone would become the 'de facto' owner of your NPM package (and potentially do bad things), so I'd recommend you learning and doing the npm publish yourself.

The syntax I've listed is an npm client workaround that pulls from git HEAD (unversioned) instead of from the npm servers.

**Comment by steren** [comment]
the library is on npm: https://www.npmjs.com/package/@google-cloud/error-reporting

What this issue is about is to provide a JS module in a more JS idiomatic way.

**Comment by mikelambert** [comment]
Ahhh sorry, I totally misunderstood, apologies for the basic reply above. I'm using it in an idiomatic way at the moment with my webpack build, so I think I assumed that this wasn't about that. But I pull from my own GitHub repo (with that pull request I just sent in)...thats why I was assuming this didn't exist on npm.

**Comment by iozeen** [comment]
@mikelambert doesn't work for me, throws error:
`Uncaught Error: make sure you loaded “dist/stackdriver-errors-concat.js” or “dist/stackdriver-errors-concat.min.js”, or that you imported the “stacktrace-js” module`
But I want to import it via ES6 and use it as util module in my react application

**Comment by Jastrzebowski** [comment]
@iozeen Check @ties [answer](https://github.com/GoogleCloudPlatform/stackdriver-errors-js/issues/15#issuecomment-306594058) in #15, not nice but working. We should work on the module, I would suggest UMD (like stacktrace-js)

**Comment by patelnav** [comment]
@Jastrzebowski : Not sure when a PR that fixes this will hit the main branch, but the shenanigans fix to this:
```
import * as StackTrace from 'stacktrace-js';
window.StackTrace = StackTrace
```
